### PR TITLE
account for grid spacing before downloading wm

### PR DESCRIPTION
**Pull request by bbuzz31**
#518

### CODE REVIEW

File: tools/RAiDER/cli/raider.py [hunk 1]
```diff
@@ -209,8 +211,8 @@ def calcDelays(iargs=None):
     aoi = params['aoi']
     model = params['weather_model']
 
-    # add a small buffer
-    aoi.add_buffer(buffer = 1.5 * model.getLLRes())
+    # adjust user requested AOI by grid size and buffer slightly
+    aoi.add_buffer(model.getLLRes())
```
Comment (dbekaert) [inline tools/RAiDER/cli/raider.py:215]: I am unclear why you modify this to be only 1* . Some of these resolutions are approximate and ensuring its >1 but technically <2 would make sure there is at least a grid cell added.

File: tools/RAiDER/cli/raider.py [hunk 2]
```diff
@@ -97,7 +97,7 @@ def read_template_file(fname):
                 raise ValueError(f"Unknown look direction {value}")
             template['look_dir'] = value.lower()
         if key == 'cube_spacing_in_m':
-            template[key] = float(value)
+            template[key] = float(value) if isinstance(value, str) else value
```
Comment (jlmaurer) [inline tools/RAiDER/cli/raider.py:100]: Is this needed? There is also the case of passing `int`s etc. If a general object is passed you don't necessarily want to allow the code to proceed.

File: tools/RAiDER/cli/raider.py (hunk 1 above)
Comment (jlmaurer) [reply tools/RAiDER/cli/raider.py:215]: I agree with @dbekaert  here. You probably want at least small amount >1.

File: tools/RAiDER/delay.py [hunk 3]
```diff
@@ -119,24 +114,20 @@ def tropo_delay(
...
-
+    out_snwe    = transform_bbox(aoi.bounds(), src_crs=4326, dest_crs=crs)
     logger.debug(f"Output SNWE: {out_snwe}")
-    logger.debug(f"Output cube spacing: {out_spacing}")
 
     # Build the output grid
     zpts = np.array(heights)
-    xpts = np.arange(out_snwe[2], out_snwe[3] + out_spacing, out_spacing)
-    ypts = np.arange(out_snwe[1], out_snwe[0] - out_spacing, -out_spacing)
+    xpts = np.arange(out_snwe[2], out_snwe[3] + aoi._output_spacing, aoi._output_spacing)
```
Comment (jlmaurer) [inline tools/RAiDER/delay.py:129]: It's not generally good practice to have `*._*` objects used by code outside the object. What I normally do is define a method e.g. "spacing()" that returns the underscored value.

File: tools/RAiDER/cli/raider.py (hunk 1 above)
Comment (bbuzz31) [reply tools/RAiDER/cli/raider.py:215]: sorry this was a typo

**Review by jlmaurer — APPROVED** [review]
@bbuzz31 this seems fine to me now. Are you still waiting on anything?

**Comment by jlmaurer** [comment]
@bbuzz31 @dbekaert looks like there is a small bug to fix, but otherwise I think this should be fine.

**Comment by bbuzz31** [comment]
> @bbuzz31 this seems fine to me now. Are you still waiting on anything?

Nope i think we should be good after the tests run. Thanks @jlmaurer

File: test/test_synthetic.py [hunk 4]
```diff
@@ -176,8 +176,12 @@ def __post_init__(self):
 
         self.wmObj = modelName2Module(self.wmName.upper().replace("-", ""))[1]()
 
+        self.hgt_lvls        = np.arange(-500, 9500, 500)
+        self._cube_spacing_m = 10000.
+
         aoi = BoundingBox(self.SNWE)
-        aoi.add_buffer(buffer = 1.5 * self.wmObj.getLLRes())
+        aoi._cube_spacing_m = self._cube_spacing_m
+        aoi.add_buffer(self.wmObj.getLLRes())
```
Comment (bbuzz31) [inline test/test_synthetic.py:184]: Note that the 1.5 * is now done within the `add_buffer` method itself

**Pull request merged**